module/VuFind/src/VuFind/Autocomplete/Solr.php

```diff
@@ -140,7 +147,13 @@ public function setConfig($params)
             $params[2] : null;
         $this->filters = [];
         if (count($params) > 3) {
-            for ($x = 3; $x < count($params); $x += 2) {
+            if (ctype_digit($params[3])) {
+                $this->limit = (int)$params[3];
+                $fs = 4;
+            } else {
+                $fs = 3;
+            }
+            for ($x = $fs; $x < count($params); $x += 2) {
```

Comment:
Two suggestions:

1.) Since we define setLimit, maybe we should use it here for consistency.

2.) A more descriptive variable name than $fs might make the code more clear.

```suggestion
            if (ctype_digit($params[3])) {
                $this->setLimit((int)$params[3]);
                $filterStartIndex = 4;
            } else {
                $filterStartIndex = 3;
            }
            for ($x = $filterStartIndex; $x < count($params); $x += 2) {
```